packages/valory/skills/abstract_round_abci/tests/test_tools/test_base.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Tests for abstract_round_abci/test_tools/base.py"""
+
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, cast
+
+import pytest
+from aea.helpers.base import cd
+from aea.mail.base import Envelope
+from aea.test_tools.utils import copy_class
+
+from packages.valory.connections.ledger.connection import (
+    PUBLIC_ID as LEDGER_CONNECTION_PUBLIC_ID,
+)
+from packages.valory.protocols.contract_api.message import ContractApiMessage
+from packages.valory.protocols.ledger_api.message import LedgerApiMessage
+from packages.valory.skills.abstract_round_abci.base import AbciAppDB, _MetaPayload
+from packages.valory.skills.abstract_round_abci.behaviours import BaseBehaviour
+from packages.valory.skills.abstract_round_abci.test_tools.base import (
+    DummyContext,
+    FSMBehaviourBaseCase,
+)
+from packages.valory.skills.abstract_round_abci.tests.data.dummy_abci import (
+    PATH_TO_SKILL,
+    PUBLIC_ID,
+)
+from packages.valory.skills.abstract_round_abci.tests.data.dummy_abci.behaviours import (
+    DummyRoundBehaviour,
+)
+from packages.valory.skills.abstract_round_abci.tests.data.dummy_abci.models import (
+    SharedState,
+)
+from packages.valory.skills.abstract_round_abci.tests.data.dummy_abci.rounds import (
+    Event,
+    SynchronizedData,
+)
+
+
+class TestFSMBehaviourBaseCaseSetup:
+    """test TestFSMBehaviourBaseCaseSetup setup"""
+
+    test_cls: FSMBehaviourBaseCase
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup class"""
+        cls.old_value = _MetaPayload.transaction_type_to_payload_cls.copy()  # type: ignore
+        _MetaPayload.transaction_type_to_payload_cls.clear()
+
+    @classmethod
+    def teardown_class(cls) -> None:
+        """Teardown class"""
+        _MetaPayload.transaction_type_to_payload_cls = cls.old_value  # type: ignore
```

Comment:
Why is this necessary? Are you doing this to avoid interference?

Reply:
yes. We do it all over the place. I made an issue out of this to be addressed during refactoring #1501